operators/src/engine/query_processor.rs

```diff
@@ -111,44 +70,32 @@ pub trait PlotQueryProcessor: Sync + Send {
 }
 
 #[async_trait]
-impl<T> QueryProcessor for Box<dyn QueryProcessor<Output = T>> {
-    type Output = T;
-    async fn query<'a>(
-        &'a self,
-        query: QueryRectangle,
-        ctx: &'a dyn QueryContext,
-    ) -> Result<BoxStream<'a, Result<Self::Output>>> {
-        self.as_ref().query(query, ctx).await
-    }
-}
-
-#[async_trait]
-impl<T> QueryProcessor for Box<dyn RasterQueryProcessor<RasterType = T>>
+impl<T> RasterQueryProcessor for Box<dyn RasterQueryProcessor<RasterType = T>>
 where
     T: Pixel,
 {
-    type Output = RasterTile2D<T>;
+    type RasterType = T;
 
-    async fn query<'a>(
+    async fn raster_query<'a>(
         &'a self,
-        query: QueryRectangle,
+        query: RasterQueryRectangle,
         ctx: &'a dyn QueryContext,
-    ) -> Result<BoxStream<'a, Result<Self::Output>>> {
+    ) -> Result<BoxStream<'a, Result<RasterTile2D<Self::RasterType>>>> {
         self.as_ref().raster_query(query, ctx).await
     }
 }
 
 #[async_trait]
-impl<V> QueryProcessor for Box<dyn VectorQueryProcessor<VectorType = V>>
+impl<V> VectorQueryProcessor for Box<dyn VectorQueryProcessor<VectorType = V>>
 where
     V: 'static,
 {
-    type Output = V;
-    async fn query<'a>(
+    type VectorType = V;
+    async fn vector_query<'a>(
```

Comment:
same here

Reply:
why do we need it?

Reply:
but what is the benefit of using a common query method if both the input and the output types are different?